core/services/vrf/v2/listener_v2.go

```diff
@@ -1124,7 +1126,9 @@ func (lsn *listenerV2) processRequestsPerSubHelper(
 				requestID := common.BytesToHash(p.req.req.RequestID().Bytes())
 				coordinatorAddress := lsn.coordinator.Address()
 				requestTxHash := p.req.req.Raw().TxHash
-				transaction, err = lsn.txm.CreateTransaction(txmgr.TxRequest{
+				ctx, cancel := lsn.chStop.NewCtx()
+				defer cancel()
```

Comment:
I wonder if defering this from inside of the tx closure could have weird side-effects :thinking: 
It may not matter, but you could move this up before the `.Transaction()` call, or even before the enclosing `for` loop, to be reused in each one.

Reply:
I noticed there's a context passed in to this method that I can use so I can actually remove creating a new one here